### ISSUE TITLE
Mention Custom Hooks for Redux Integration Docs

### ISSUE THREAD

**Issue by ricokahler** [ReactTraining/react-router#6972]
I was reading through these [Redux Integration](https://reacttraining.com/react-router/web/guides/redux-integration) docs when I had a bit of a realization:

Now that there are custom hooks for both `react-router` _and_ `react-redux`, the new preferred way to integrate redux and react-router is should be a custom hook 🎉.

**This issue exists as a feature request to add this information to the docs.**

I would be willing to work on that if I have your blessing.

Example:

```js
import { useSelector } from 'react-redux';
import { useRouteMatch } from 'react-router';

function useCurrentMovie() {
  const currentMovieMatch = useRouteMatch('/movies/:movieId');
  const movieId = currentMovieMatch?.params?.movieId;
  const movie = useSelector(state => state.movies[movieId]);
  return movie;
}
```

Let me know what you think!

**Comment by timdorr** [comment]
There are known disadvantages to React Redux's hooks, namely the loss of top-down updates. We still recommend `connect` for most use cases.

The main point of the article is pointing out the problem of blocked updates. I don't think we need to expand its scope at this time.

**Comment by ricokahler** [comment]
> There are known disadvantages to React Redux's hooks, namely the loss of top-down updates.

I thought issues related to top-down updates were mostly fixed with batched updates. Am I missing something?

> The main point of the article is pointing out the problem of blocked updates. I don't think we need to expand its scope at this time.

I also thought blocked updates was fixed by updating to the new React context.

---

I would argue repurposing this doc might help a lot of people. I found that custom hooks are great for doing joins from global sources and redux and react-router are that.
